app/src/main/java/com/example/tasky/feature_agenda/data/repository/ReminderRepositoryImpl.kt

```diff
@@ -0,0 +1,100 @@
+package com.example.tasky.feature_agenda.data.repository
+
+import android.os.Build
+import androidx.annotation.RequiresApi
+import com.example.tasky.feature_agenda.data.local.AgendaDatabase
+import com.example.tasky.feature_agenda.data.mapper.toReminder
+import com.example.tasky.feature_agenda.data.mapper.toReminderDto
+import com.example.tasky.feature_agenda.data.mapper.toReminderEntity
+import com.example.tasky.feature_agenda.data.remote.TaskyAgendaApi
+import com.example.tasky.feature_agenda.domain.model.AgendaItem
+import com.example.tasky.feature_agenda.domain.repository.ReminderRepository
+import com.example.tasky.util.Result
+import retrofit2.HttpException
+import java.io.IOException
+
+class ReminderRepositoryImpl(
+    private val api: TaskyAgendaApi,
+    private val db: AgendaDatabase
+): ReminderRepository {
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    override suspend fun createReminder(reminder: AgendaItem.Reminder): Result<Unit> {
+
+        return try {
+
+            api.createReminder(reminder.toReminderDto())
+
+            val reminderEntity = reminder.toReminderEntity()
+            db.reminderDao.upsertReminder(reminderEntity)
```

Comment:
Same as in the last PR, this needs to go before making the API call to be able to use the app in offline mode. Then try to sync it later